spec/controllers/v2/concerns/rendering_spec.rb

```diff
@@ -54,6 +54,26 @@
     end
   end
 
+  context 'under invalid parent hierarchy', if: parents.count > 1 do
+    (1..parents.length - 1).each do |count|
+      parents.combination(count).each do |missing_parents|
```

Comment:
@RoamingNoMaD would `invalid_parents` be better name?

```suggestion
      parents.combination(count).each do |invalid_parents|
```